pkg/security/controller/namespace_scc_allocation_controller.go

```diff
@@ -216,6 +220,35 @@ func (c *NamespaceSCCAllocationController) allocate(ns *corev1.Namespace) error
 	return nil
 }
 
+func (c *NamespaceSCCAllocationController) deallocate(ns *corev1.Namespace) error {
```

Comment:
This is insufficient.  If this controller is stopped while a namespace is being deleted, then no one will ever fix it and eventually you will leak enough indices that you run out of space.

You MUST rebuild the state of the world periodically and do a read, check, replace.  If you race with another instance of your controller the update method will fail and you just reenqueue the repair.

Please use repair correctly - repair is safe if Get -> Update does not return a 409.

Reply:
> This is insufficient. If this controller is stopped while a namespace is being deleted, then no one will ever fix it and eventually you will leak enough indices that you run out of space.

If it's stopped it will call `Repair` upon restart, in which case it should fix the current state of the world. I'm not removing the `Repair` I'm just adding ad-hoc deallocation. 


> You MUST rebuild the state of the world periodically and do a read, check, replace. If you race with another instance of your controller the update method will fail and you just reenqueue the repair.

Yeah, that's fair. I didn't want to call `Repair` on every deletion since that might be taking too long, but it's definitely worthy to call this periodically.